src/framework/components/collision/system.js

```diff
@@ -407,9 +411,11 @@ class CollisionMeshSystemImpl extends CollisionSystemImpl {
 
         const triMeshShape = new Ammo.btBvhTriangleMeshShape(triMesh, true /* useQuantizedAabbCompression */);
 
-        const scaling = system._getNodeScaling(node);
-        triMeshShape.setLocalScaling(scaling);
-        Ammo.destroy(scaling);
+        if (!scale) {
```

Comment:
I wonder if instead of passing the scale parameter to the function, and then either using it, or extracting it from the node, we should not pass it in, and always extract it from the node? And apply directly to vertices that you introduced here. I'm just not sure what the advantage is in having to pass it in, considering we can extract it from the node?

Reply:
Hmm, in case of Render Asset that node is a temp node, which will always has a scale 1, so not needed. In case of Mesh Asset it will be a node from the mesh instance. I wasn't sure if nodes on model mesh instances could have a different scale, so left it there as is.